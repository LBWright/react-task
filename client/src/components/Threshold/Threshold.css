.form-control {
  width: 100%;
  display: flex;
  font-size: 14px;
}
.threshold-container {
  margin: 20px;
  border-radius: 5px;
  border: 1px solid LightGray;
  height: 200px;
  width: 100%;
  padding: 10px;
}
input {
  font-size: inherit;
  height: 40px;
  width: 80%;
  border-radius: 5px;
  border: 1px solid LightGray;
  padding: 0 5px;
}

button {
  font-size: inherit;

  height: 40px;
  width: 45px;
  color: white;
  background: rgb(136, 132, 216);
  border-radius: 5px;
  padding: 0 5px;
  border: none;
  margin: 5px;
  cursor: pointer;
}

.threshold-reporter {
  font-size: 20px;
}
