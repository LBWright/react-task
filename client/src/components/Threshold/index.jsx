import React, { useState } from 'react'
import { func, number } from 'prop-types'

function Threshold({ setThreshold, threshold }) {
  const [input, setInput] = useState('')

  function handleSubmit(event) {
    event.preventDefault()
    setThreshold(input)
  }
  return (
    <div style={{ margin: 20 }}>
      <h2>Threshold Controls</h2>
      <form onSubmit={handleSubmit}>
        <input
          placeholder="Set Alert Threshold"
          value={input}
          onChange={({ target: { value } }) => setInput(value)}
          type="number"
          onBlur={handleSubmit}
          defaultValue={threshold}
        />
        <button type="submit">Set Threshold</button>
      </form>
      <div>
        {threshold ? (
          <p>Current Alert Threshold: {threshold}</p>
        ) : (
          <p>No Alert Threshold Specified</p>
        )}
      </div>
    </div>
  )
}

Threshold.propTypes = {
  setThreshold: func.isRequired,
  threshold: number,
}

export default Threshold
