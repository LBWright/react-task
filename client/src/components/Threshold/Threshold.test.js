import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Threshold from '.'

describe('<Threshold />', () => {
  it('should render without crashing', () => {
    // Props and mount are scoped to keep jest.fn() accurate
    const mockProps = {
      setThreshold: jest.fn(),
      threshold: 15,
    }
    const threshold = mount(<Threshold {...mockProps} />)

    expect(threshold).toBeTruthy()
  })

  it('should handle user input on submit', () => {
    const mockProps = {
      setThreshold: jest.fn(),
      threshold: 15,
    }
    const threshold = mount(<Threshold {...mockProps} />)
    const userInput = threshold.find('#threshold-input')
    const thresholdForm = threshold.find('#threshold-form')
    userInput.simulate('change', { target: { value: 25 } })
    thresholdForm.simulate('submit')
    expect(mockProps.setThreshold).toHaveBeenCalledTimes(1)
    expect(mockProps.setThreshold).toHaveBeenCalledWith(25)
  })

  it('should submit user input button click', () => {
    const mockProps = {
      setThreshold: jest.fn(),
      threshold: 15,
    }
    const threshold = mount(<Threshold {...mockProps} />)
    const userInput = threshold.find('#threshold-input')
    const thresholdButton = threshold.find('#threshold-submit')
    userInput.simulate('change', { target: { value: 40 } })
    thresholdButton.simulate('click')
    expect(mockProps.setThreshold).toHaveBeenCalledTimes(1)
    expect(mockProps.setThreshold).toHaveBeenCalledWith(40)
  })
  it('should match snapshot', () => {
    const mockProps = {
      setThreshold: jest.fn(),
      threshold: 15,
    }
    const thresholdSnap = renderer.create(<Threshold {...mockProps} />)
    expect(thresholdSnap.toJSON()).toMatchSnapshot()
  })
})
