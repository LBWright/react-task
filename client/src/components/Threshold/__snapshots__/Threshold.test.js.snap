// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Threshold /> should match snapshot 1`] = `
<div
  style={
    Object {
      "margin": 20,
    }
  }
>
  <h2>
    Threshold Controls
  </h2>
  <form
    id="threshold-form"
    onSubmit={[Function]}
  >
    <input
      id="threshold-input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Set Alert Threshold"
      type="number"
      value=""
    />
    <button
      id="threshold-submit"
      onClick={[Function]}
      type="submit"
    >
      Set Threshold
    </button>
  </form>
  <div>
    <p>
      Current Alert Threshold: 
      15
    </p>
  </div>
</div>
`;
